src/lightning/__setup__.py

```diff
@@ -115,7 +115,7 @@ def _setup_args() -> Dict[str, Any]:
         "python_requires": ">=3.8",  # todo: take the lowes based on all packages
         "entry_points": {
             "console_scripts": [
-                "lightning = lightning.app.cli.lightning_cli:main",
+                "lightning = lightning:_cli_entry_point",
```

Comment:
What's the advantage of having a function at the root instead of linking to the file's `main`?

Reply:
I can't put it under lightning.app, because any import path under it would require dependencies that aren't installed. The root of the source seemed to be the only place where I can put it so that it's accessible by all packages.